yarpl/include/yarpl/Refcounted.h

```diff
@@ -273,13 +273,26 @@ Reference<T> make_ref(Args&&... args) {
   return Reference<T>(new T(std::forward<Args>(args)...));
 }
 
-template <typename T>
+template <
+    typename T,
+    typename =
+    std::enable_if_t<std::is_base_of<Refcounted, std::decay_t<T>>::value>>
 Reference<T> get_ref(T& object) {
+  assert(
```

Comment:
DCHECK_GT depends on including gtest related header. I did not want to include that in this header as this is a low level header.
We all use `assert` in this file.